crates/oxc_linter/src/rules/eslint/default_param_last.rs

```diff
@@ -0,0 +1,118 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(default-param-last): Default parameters should be last")]
+#[diagnostic(severity(warning), help("Enforce default parameters to be last."))]
+struct DefaultParamLastDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct DefaultParamLast;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforce default parameters to be last
+    ///
+    /// ### Why is this bad?
+    /// Putting default parameter at last allows function calls to omit optional tail arguments.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Correct: optional argument can be omitted
+    /// function createUser(id, isAdmin = false) {}
+    /// createUser("tabby")
+    ///
+    /// // Incorrect: optional argument can **not** be omitted
+    /// function createUser(isAdmin = false, id) {}
+    /// createUser(undefined, "tabby")
+    /// ```
+    DefaultParamLast,
+    style
+);
+
+impl Rule for DefaultParamLast {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::Function(function) => {
+                if !function.is_declaration() & !function.is_expression() {
+                    return;
+                }
+                let mut has_seen_plain_param = false;
+                for i in (0..function.params.items.len()).rev() {
```

Comment:
```suggestion
                for param in function.params.items.iter().rev() {
```

We don't need to get the param via the index